### DISCUSSION TITLE
Help：question and confusion after finishing the installation of MIMIC-IV-2.2

### DISCUSSION THREAD

**Discussion by washingtonwuyi** [MIT-LCP/mimic-code#1624] [MIMIC-IV] [answered]
Hello guys!Here is the problem description:after completing the installation of MIMIC-IV-2.2 via SQL Shell,I did the test of my build and it seemed that everything went the right way.However,next time I reentered the SQL Shell and retested my build,it came out with an error. Where have I missed?Any solutions?Thanks！
![aa3c106d8151cbd4a5cce25996a97b4](https://github.com/MIT-LCP/mimic-code/assets/136318775/cca556b3-575e-4dba-97c2-496371024bc6)
![4600b30b9c2acbff100ae697d35c4f7](https://github.com/MIT-LCP/mimic-code/assets/136318775/106690ee-9f83-416c-b287-a5724372b411)

**Comment by heisenbug-1** [comment]
what happens when you run SHOW SEARCH_PATH; ?

**Reply by washingtonwuyi** [reply]
the fist time it was OK but the second i reentered the postgres and typed the same search command it came out with an error

**Reply by heisenbug-1** [reply]
Okay, I see. But what do you get when you type in SHOW SEARCH_PATH;? Do you see the mimic database?

**Reply by alistairewj** [reply]
You might be excluding the semi-colon `;` required at the end of the SHOW statement

**Reply by washingtonwuyi** [reply]
Thank you guys for your replies and your concern.Actually I'm new to the codes,github,and of course the psql ~so I followed the MIMIC official instructions which can be found here：https://mimic.mit.edu/docs/gettingstarted/local/install-mimic-locally-windows/，and today I've tried it again.I installed the MIMIC-IV-2.2 into the psql,then verified my build by using the official instructions like this:"select icustay_id, intime, outtime from icustays limit 10;" I did it immediately after finishing the installation of the database without close the SQL Shell,and it came out with results,which went so normal.Then I restarted the SQL Shell(closed it then reopened it),did the same verification by using the same command "select icustay_id, intime, outtime from icustays limit 10;" It came out with an error.
![微信图片_20230815215310](https://github.com/MIT-LCP/mimic-code/assets/136318775/a0e4ecb5-78ec-41da-8337-3c534386a467)
![微信图片_20230815215325](https://github.com/MIT-LCP/mimic-code/assets/136318775/79eba224-2b07-4332-823d-1beb72b6e239)

**Reply by washingtonwuyi** [reply]
This is what I've got after typing in SHOW SEARCH_PATH;
![微信图片_20230815220254](https://github.com/MIT-LCP/mimic-code/assets/136318775/4d47e3cc-cc00-4938-bf95-d2658cea1eaa)

**Reply by tompollard** [reply, marked as answer]
@washingtonwuyi I would strongly suggest working with the instance of the database on Google BigQuery, unless you need to be hosting it locally. Instructions for working in the cloud are at: https://mimic.mit.edu/docs/gettingstarted/cloud/

If you need to continue with a locally hosted version of the database, I would begin by following a psql tutorial. From a quick Google, https://tomcam.github.io/postgres/ looks like it covers the commands you need.

Basically it looks like the database engine can't find the `icustays` table specified in your command. Usually you see this kind of error if either:

1. the current user doesn't have permission to see/query the table or
2. the table is not available on the current search path

I think your problem is 2. To fix this, you need to locate the table that you're interested in, and then re-run your query. It's been a while since I used postgres, but from the top of my head this might help:

- Begin with `\l` to list the databases. 
- Connect to the database containing the data with `\c <name>`
- List the schema on this database with `\dn`
- Set your search path to the appropriate schema with `set search_path to <schema>`
- You can view a list of tables with `\dt`
- If the schema contains the `icustays` table, your query will run.

**Reply by washingtonwuyi** [reply]
Thank you, guys! Problem solved! There are so many warm-hearted people here! I truly appreciate the help and suggestions from all of you. Thanks to the MIMIC team and the community, which is pretty good and cool, I've received a lot of support and gained valuable experience.
![图片](https://github.com/MIT-LCP/mimic-code/assets/136318775/6b32c684-f4e5-4bf4-9595-7b9cf29e19fc)
